AzureCalling/Controllers/AudioDeviceSelectionViewController.swift

```diff
@@ -0,0 +1,131 @@
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        openDeviceTable()
+    }
+
+    func createDeviceTable() {
+        deviceTable = UITableView()
+        deviceTable.isHidden = true
+        deviceTable.allowsSelection = true
+        deviceTable.isScrollEnabled = false
+        deviceTable.layer.cornerRadius = 8
+        deviceTable.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(deviceTable)
+        let audioDeviceCell = UINib(nibName: "BottomDrawerCellView",
+                                      bundle: nil)
+        deviceTable.register(audioDeviceCell, forCellReuseIdentifier: "BottomDrawerCellView")
+        deviceTable.dataSource = self
+        deviceTable.delegate = self
+        deviceTable.reloadData()
+
+        var deviceTableConstraints = [
+        deviceTable.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+        deviceTable.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+        deviceTable.heightAnchor.constraint(equalToConstant: 100)
+        ]
+
+        let hideConstraint = deviceTable.topAnchor.constraint(equalTo: view.bottomAnchor)
+        hideConstraint.priority = .defaultLow
+        deviceTableConstraints.append(hideConstraint)
+
+        NSLayoutConstraint.activate(deviceTableConstraints)
+    }
+
+    func openDeviceTable() {
+        deviceTable.isHidden = false
+
+        let centerYConstraint = NSLayoutConstraint(item: deviceTable!,
+                attribute: .bottom,
+                relatedBy: .equal,
+                toItem: self.view,
+                attribute: .bottom,
+                multiplier: 1,
+                constant: 0)
+        centerYConstraint.priority = .required
+        self.view.addConstraint(centerYConstraint)
+        UIView.animate(withDuration: 0.15, delay: 0, options: .beginFromCurrentState, animations: {
+            self.view.layoutIfNeeded()
+        }, completion: nil)
+    }
+
+    func createAudioDeviceOptions() {
+        let audioDeviceTypes = AudioSessionManager.getAllAudioDeviceTypes()
+        let currentAudioDeviceType = AudioSessionManager.getCurrentAudioDeviceType()
+
+        for audioDeviceType in audioDeviceTypes {
+            let audioDeviceOption = AudioDeviceOption(image: audioDeviceType.image, name: audioDeviceType.name, accessoryImage: audioDeviceType.accessoryImage, enabled: audioDeviceType == currentAudioDeviceType)
+            audioDeviceOptions.append(audioDeviceOption)
+        }
+    }
+
+    // MARK: UITableViewDelegate events
+
+    func tableView(_ tableView: UITableView,
+                   didSelectRowAt indexPath: IndexPath) {
+        let cell = tableView.cellForRow(at: indexPath) as! BottomDrawerCellView
+        let audioDeviceType = AudioDeviceType(rawValue: cell.title.text)!
+        AudioSessionManager.switchAudioDeviceType(audioDeviceType: audioDeviceType)
+        dismissSelf(sender: self)
```

Comment:
For `didSelectRowAt` at tableView. For the selected data source, we usually get access it by 
```
let selectedItem = audioDeviceOptions[indexPath]
```
rather than get the selected data by using the cellView. 
See example:
https://developer.apple.com/documentation/uikit/uitableviewdelegate/handling_row_selection_in_a_table_view?language=objc